islpy/oppool.py

```diff
@@ -0,0 +1,830 @@
+__copyright__ = "Copyright (C) 2021 Kaushik Kulkarni"
```

Comment:
Not loving the name. The main advance here is that we're "taking over" in terms of naming/labeling things from isl. Better caching (is that what you mean by pooling?) is just a side effect of that. So I'd prefer a name that reflects that. `int_names`? `ext_naming`? `sep_naming`? ...